src/ORM/ArrayList.php

```diff
@@ -159,6 +160,21 @@ public function toNestedArray()
      */
     public function limit($length, $offset = 0)
     {
+        // Type checking: designed for consistency with DataList::limit()
+        if (!is_numeric($length) || !is_numeric($offset)) {
+            Deprecation::notice(
+                '4.0',
+                'Arguments to ArrayList::limit() should be numeric'
+            );
+        }
+
+        if ($length < 0 || $offset < 0) {
+            Deprecation::notice(
+                '4.0',
```

Comment:
We should probably deprecate from 4.3 on

Reply:
I thought the first arg was the target version for removal, in this case `'5.0'`?

Reply:
No it's the version the notice will start being raised from